From 95946cdf47b48fe8d0208e5703c086bfed3e9fe7 Mon Sep 17 00:00:00 2001
From: Wally Gleemer <Wally.Gleemer@windriver.com>
Date: Tue, 25 Mar 2008 16:21:34 -0700
Subject: [PATCH] cavium: fix out of tree builds for ethernet.

Trying to build with the objs out and separate from the source tree
with "make O=../objdir" fails on this cavium-ethernet Makefile.`

Signed-off-by: Wally Gleemer <Wally.Gleemer@windriver.com>
---
 drivers/net/cavium-ethernet/Makefile |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/net/cavium-ethernet/Makefile b/drivers/net/cavium-ethernet/Makefile
index a4836c1..66a0ca3 100644
--- a/drivers/net/cavium-ethernet/Makefile
+++ b/drivers/net/cavium-ethernet/Makefile
@@ -40,7 +40,11 @@
 # OF USE OR PERFORMANCE OF THE SOFTWARE LIES WITH YOU.
 
 # Common flags to be passed for driver compilation
-EXTRA_CFLAGS += -Winline -Wall -I arch/mips/cavium-octeon/gpl-executive/config -I ${OCTEON_ROOT}/executive
+OCTEON_ROOT = $(srctree)/arch/mips/cavium-octeon
+
+EXTRA_CFLAGS += -Winline -Wall \
+	-I$(OCTEON_ROOT)/gpl-executive/config \
+	-I$(OCTEON_ROOT)/executive
 
 obj-${CONFIG_CAVIUM_ETHERNET} :=  cavium-ethernet.o
 
-- 
1.5.5.1

